examples/mini-hlint/src/Main.hs

```diff
@@ -98,9 +98,19 @@ main = do
       s <- readFile' file
       let flags = defaultDynFlags fakeSettings fakeLlvmConfig
       case parse file flags s of
-        POk _ m -> analyzeModule flags m
-        PFailed s -> sequence_
-          [ putStrLn $ showSDoc flags msg
-          | msg <- pprErrMsgBagWithLoc $ snd $ getMessages s flags
-          ]
+        PFailed s ->
+          report flags (snd (getMessages s flags))
+        POk s m -> do
+          let (warns, errs) = getMessages s flags
+          if not (null errs)
+            then report flags errs
+            else do
+              unless (null warns) $ report flags warns
```

Comment:
Why not just `report $ errs ++ warns`? On `[]` report does nothing.

Reply:
(1) We agree we will not proceed to `analyzeModule` if there are (so called "non-fatal") errors.
(2) Nice ideas on reporting but `warns` and `errs` can't be combined that way (not lists). Next iteration though will unconditionally report warnings in the `POk` case though.